crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1331,6 +1331,62 @@ impl<'src> Parser<'src> {
 
         self.expect(TokenKind::FStringEnd);
 
+        // test_ok pep701_f_string_py312
+        // # parse_options: {"target-version": "3.12"}
+        // f'Magic wand: { bag['wand'] }'     # nested quotes
+        // f"{'\n'.join(a)}"                  # escape sequence
+        // f'''A complex trick: {
+        //     bag['bag']                     # comment
+        // }'''
+        // f"{f"{f"{f"{f"{f"{1+1}"}"}"}"}"}"  # arbitrary nesting
+
+        // test_err pep701_f_string_py311
+        // # parse_options: {"target-version": "3.11"}
+        // f'Magic wand: { bag['wand'] }'     # nested quotes
+        // f"{'\n'.join(a)}"                  # escape sequence
+        // f'''A complex trick: {
+        //     bag['bag']                     # comment
+        // }'''
+        // f"{f"{f"{f"{f"{f"{1+1}"}"}"}"}"}"  # arbitrary nesting
+
+        // the inner variant here doesn't matter, just checking if PEP 701 f-strings are supported
+        if UnsupportedSyntaxErrorKind::Pep701FString(FStringKind::NestedQuote)
+            .is_unsupported(self.options.target_version)
+        {
+            let quote_str = flags.quote_str();
+            for expr in elements.expressions() {
+                if let Some(slash_index) = self.source[expr.range].find('\\') {
```

Comment:
I think this also gives you false positives for

```py
>>> f"test \
... more"
'test more'
```

Reply:
ah no, I think this should be fine because you only iterate over expressions. It might still be worth adding a test for it as well as for 

```
f"test {a \
    } more"
```